### PR TITLE
Update of pom.xml and TeamFoundationServerScm.java to support "NO_CI" flag

### PR DESCRIPTION
**Pull request by dbascans**
- Add the ability of avoid the scm trigger when the check-in comment has the flag "NO_CI" in the comment

### CODE REVIEW

**Comment by olivierdagenais** [comment]
Sorry for the very long delay; when I first got commit access to this project, I had been only tackling my immediate needs at $DAYJOB.

I am planning to review these changes for release 3.3.0.

File: pom.xml [hunk 1]
```diff
@@ -46,27 +61,7 @@
         </configuration>
       </plugin>
 
-      <plugin>
```
Comment (mosabua) [inline pom.xml]: Imho this should not be removed. If anything it should be updated to java 1.6 or 1.7 if thats what you want to support. But NOT remove. Also it should be in a separate pull request since this is not related to the actual change.

File: pom.xml [hunk 2]
```diff
@@ -1,16 +1,15 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <parent>
-    <groupId>org.jvnet.hudson.plugins</groupId>
```
Comment (mosabua) [inline pom.xml]: Are these changes required for this PR? If not they should be removed.

**Comment by mosabua** [comment]
This would have to be rebased and changed quite a bit. At this stage I am not sure about the value of this feature in any case and therefore I am closing this PR. It seems weird to control CI server builds via commit messages. Feel free to submit a new PR based on the latest codebase that re-implements the feature.
